enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+installModes:
+- type: OwnNamespace
+  supported: true
+- type: SingleNamespace
+  supported: true
+- type: MultiNamespace
+  supported: true
+- type: AllNamespaces
+  supported: true
+icon: my-icon-png-base64-encoded
+labels:
+  operated-by: packageserver
+```
+
+#### `opm` Support
+
+[`opm`](https://github.com/operator-framework/operator-registry#overview) must be updated to provide support for building `k8s+v1` bundles. This can be acheived with explicit flags or through CSV/olm.yaml file presence.
+
+### CSV Synthesis
+
+OLM will support `k8s+v1` bundles by synthesizing an CSV from its contents. Resources usually generated by a CSV (e.g. APIService and Service) will not be directly applied, but instead be projected onto a synthesized CSV.
+
+This intermediary step (creating a CSV internally to represent the kubernetes objects) is a delibrate decision in 4.5 to help focus and deliver the feature. This may be relaxed in future release versions.
+
+### Indexes
+
+When a `k8s+v1` bundle is added to an index, `opm` will also synthesize and add a CSV. A lot of the existing `opm` and `operator-registry` code assumes bundles always contain CSVs, which make synthesis the simplest path to supporting `k8s+v1` bundles -- in the future, we may want to revisit this decision, especially when the [`Operator` API](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#proposal) is released.
+
+To fill out the dependency graph between versions of bundles, OLM considers the "replaces" field on the CSV which points to the previous version of the operator. This field will need to be supplied in the additional metadata the operator author provides in the bundle so that OLM can build dependency graphs. There is another enhancement around semantic versioning that would remove this requirement altogether, but it is independent of this enhancement.
+
+Supporting indexes with different mediatypes is a key requirement. Both `k8s+v1`  and `registry+v1` type bundles should be able to be added to an existing index. There is the potential of versioning the index itself to help differentiate between newer index images, that include the `k8s+v1` mediatype, and older versions.
+
+#### Resolving, Unpacking, and Applying
+
+At install, a CSV must be generated for a `k8s+v1` bundle so that OLM can manage the operator it represents on the cluster. It must also take care not to apply any bundle resources that are usually generated by a CSV (e.g. APIService and Service). This means that when OLM resolves and unpacks a bundle, it needs to recognize its mediatype and take appropriate action.
+
+#### Legacy Support for AppRegistry
+
+A bundle of type `k8s+v1` can be stored in an AppRegistry type catalog after synthesizing a CSV and removing resources that are usually generated by CSVs. This allows clusters with AppRegistry type catalogs to receive updates to operators whose authors have switched to publishing `k8s+v1` bundles.
```

Comment:
While this is nice, it is not strictly required since `appregistry` is deprecated and we actively discourage from using it.

Reply:
Correct me if I'm wrong, but we need to support this for backporting updates to existing OpenShift clusters or else this format can't be used.